### PR TITLE
wallet-ext: toasts

### PR DESCRIPTION
**Pull request by pchrysochoidis**
* use react-hot-toast
* change style from loading, error and success states
* fix faucet response type (sdk)
* use toasts for messages
* use mutation for faucet requests
* remove redux faucet slice
* use provider api for the request
* this has as a side effect to remove the remaining time to wait for the next request in case it was rate limited (since we don't get that from the api) and shows the error messages from the sdk

https://user-images.githubusercontent.com/10210143/212194653-5f44ac5b-b5bc-488d-b81a-28f704d72ec5.mov

https://user-images.githubusercontent.com/10210143/212195443-3dc61ead-e517-4531-80b5-fde0543ecf6f.mov

Latest update with extra space for toast to account for nav tabs when visible cc @mystie711 

https://user-images.githubusercontent.com/10210143/212395210-07f2361c-818d-498e-8087-6290ea0087e6.mov

closes APPS:293

### CODE REVIEW

File: apps/wallet/src/ui/index.tsx [hunk 1]
```diff
@@ -48,6 +51,38 @@ function renderApp() {
                         <QueryClientProvider client={queryClient}>
                             <ErrorBoundary>
                                 <App />
+                                <Toaster
+                                    position="bottom-center"
+                                    toastOptions={{
+                                        loading: {
+                                            icon: (
+                                                <LoadingIndicator className="!bg-steel !border-white !border-r-transparent" />
+                                            ),
```
Comment (Jibz1) [inline apps/wallet/src/ui/index.tsx:60]: not blocking Instead of overwriting `LoadingIndicator` we may want to use CVA

**Review by Jibz1 — APPROVED** [review]

File: apps/wallet/src/ui/app/shared/faucet/request-button/index.tsx [hunk 2]
```diff
@@ -26,18 +28,26 @@ function FaucetRequestButton({
     const network = useAppSelector(({ app }) => app.apiEnv);
     const networkName = API_ENV_TO_INFO[network].name;
     const showFaucetRequestButton = API_ENV.customRPC !== network;
-    const dispatch = useAppDispatch();
-    const loading = useAppSelector(({ faucet }) => faucet.loading);
+    const mutation = useFaucetMutation();
+    const isMutating = useIsFaucetMutating();
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/shared/faucet/request-button/index.tsx:32]: Why not just get this from the `useFauceMutation`?

Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/shared/faucet/request-button/index.tsx:32]: Good point, I'll change it

Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/shared/faucet/request-button/index.tsx]: Done

File: apps/wallet/src/ui/app/shared/faucet/useFaucetMutation.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            return transferred_gas_objects.reduce(
+                (total, { amount }) => total + amount,
+                0
+            );
+        },
+    });
+    return {
+        ...mutation,
+        /**
+         * is any faucet request in progress across different instances of the mutation
```
[the remark this answers is not in the document]
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/shared/faucet/useFaucetMutation.ts:33]: Unfortunately no, it seems the don't share state and every time the component that uses it mounts it creates a new instance

[the remark this answers is not in the document]
Comment (Jordan-Mysten) [reply apps/wallet/src/ui/app/shared/faucet/useFaucetMutation.ts:33]: Ah okay. Then follow-up: is this concretely a problem that we need to solve? Don't consider this discussion blocking mostly just want to understand.

[the remark this answers is not in the document]
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/shared/faucet/useFaucetMutation.ts:33]: I think that's the required behaviour, be able to only have 1 ongoing request every time. So I would say yes!

(Waiting for @mystie711 to have a look as well)

[the remark this answers is not in the document]
Comment (Jordan-Mysten) [reply apps/wallet/src/ui/app/shared/faucet/useFaucetMutation.ts:33]: Got it, yeah if it's a hard requirement then this makes sense to me.

**Comment by mystie711** [comment]
@pchrysochoidis a couple observations + questions
- "Request Sui Devnet SUI Tokens" should be "Request Devnet SUI Tokens"
- why is the banner in such an awkward place in fullscreen view? Please ping me on Slack...it looks like we need to talk through options for this and in the wallet extension window :)

**Comment by pchrysochoidis** [comment]
> @pchrysochoidis a couple observations + questions
> 
> * "Request Sui Devnet SUI Tokens" should be "Request Devnet SUI Tokens"
> * why is the banner in such an awkward place in fullscreen view? Please ping me on Slack...it looks like we need to talk through options for this and in the wallet extension window :)

@mystie711 those should be addressed now

**Comment by mystie711** [comment]
Amazing!

File: apps/wallet/tailwind.config.js [hunk 4]
```diff
@@ -28,6 +28,12 @@ module.exports = {
             width: {
                 7.5: '30px',
             },
+            borderRadius: {
+                '2lg': '0.625rem',
```
Comment (Jordan-Mysten) [inline apps/wallet/tailwind.config.js:32]: Not a blocking concern but I don't love this naming. Probably should revamp all border radiuses with abhi.

**Review by Jordan-Mysten — APPROVED** [review]

**Pull request merged**